Sources/FoundationEssentials/FileManager/FileManager+Directories.swift

```diff
@@ -207,6 +234,35 @@ extension _FileManagerImpl {
         withIntermediateDirectories createIntermediates: Bool,
         attributes: [FileAttributeKey : Any]? = nil
     ) throws {
+#if os(Windows)
+        try path.withNTPathRepresentation { pwszPath in
+            if createIntermediates {
+                var isDirectory: Bool = false
+                if fileExists(atPath: path, isDirectory: &isDirectory) {
+                    guard isDirectory else {
+                        throw CocoaError.errorWithFilePath(path, win32: ERROR_FILE_EXISTS, reading: false)
+                    }
+                    return
+                }
+
+                let parent = path.deletingLastPathComponent()
+                if !parent.isEmpty {
+                    try createDirectory(atPath: parent, withIntermediateDirectories: true, attributes: attributes)
```

Comment:
This pattern is really similar to the non-`FOUNDATION_FRAMEWORK` pattern below. Do you think there's any way to coalesce the two together into a portable solution (I think the only difference is really `withFileSystemRepresentation` vs. `withNTPathRepresentation` and `mkdir` vs. `CreateDirectoryW`). It might not be doable, but if for this one we can just if-def around those calls maybe we can make this a single implementation rather than an effective "fork"?

Reply:
I tried to do that but found myself quickly annoyed with the `#if` conditions.

Reply:
Ok that's fair if the `#if` conditions become too unruly that way then the separate block is probably cleaner - it's just nice to maintain one implementation wherever we can so that the behavior is the same

Reply:
Absolutely, I prefer to have a single path whenever possible.